pkg/stacks/thanos/alert_customization.go

```diff
@@ -0,0 +1,1027 @@
+	}
+
+	// Update the secret with the new config
+	updateCmd := []string{
+		"patch", "secret", secretName,
+		"-n", "monitoring",
+		"--type", "merge",
+		"-p", fmt.Sprintf("{\"data\":{\"alertmanager.yaml.gz\":\"%s\"}}", encodedConfig),
+	}
+
+	if _, err := utils.ExecuteCommand(ctx, "kubectl", updateCmd...); err != nil {
+		return fmt.Errorf("failed to update AlertManager secret: %w", err)
+	}
+
+	// Restart AlertManager to pick up the new config
+	restartCmd := []string{
+		"rollout", "restart", "deployment",
+		"-n", "monitoring",
+		"-l", "app.kubernetes.io/name=alertmanager",
+	}
+
+	if _, err := utils.ExecuteCommand(ctx, "kubectl", restartCmd...); err != nil {
+		return fmt.Errorf("failed to restart AlertManager: %w", err)
+	}
+
+	return nil
+}
+
+// GetEmailConfiguration extracts email configuration from AlertManager config
+func (a *AlertCustomization) GetEmailConfiguration(config string) map[string]interface{} {
+	var amConfig map[string]interface{}
+	if err := yaml.Unmarshal([]byte(config), &amConfig); err != nil {
+		return map[string]interface{}{"enabled": false}
+	}
+
+	// Get global SMTP settings
+	global, ok := amConfig["global"].(map[string]interface{})
+	if !ok {
+		return map[string]interface{}{"enabled": false}
+	}
+
+	smtpURL := ""
+	if smtpHost, exists := global["smtp_smarthost"]; exists {
+		smtpURL = fmt.Sprintf("%v", smtpHost)
+	}
+
+	from := ""
+	if smtpFrom, exists := global["smtp_from"]; exists {
+		from = fmt.Sprintf("%v", smtpFrom)
+	}
+
+	// Get receivers
+	receivers, ok := amConfig["receivers"].([]interface{})
+	if !ok {
+		return map[string]interface{}{"enabled": false}
+	}
+
+	var toAddresses []string
+	for _, r := range receivers {
+		receiver, ok := r.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		if emailConfigs, exists := receiver["email_configs"]; exists {
+			if emailConfigList, ok := emailConfigs.([]interface{}); ok {
+				for _, emailConfig := range emailConfigList {
+					if emailConfig, ok := emailConfig.(map[string]interface{}); ok {
+						if toAddr, exists := emailConfig["to"]; exists {
+							toAddresses = append(toAddresses, fmt.Sprintf("%v", toAddr))
+						}
+					}
+				}
+			}
+		}
+	}
+
+	if len(toAddresses) > 0 {
+		return map[string]interface{}{
+			"enabled":  true,
+			"smtp_url": smtpURL,
+			"from":     from,
+			"to":       strings.Join(toAddresses, ", "),
+		}
+	}
+
+	return map[string]interface{}{"enabled": false}
+}
+
+// GetTelegramConfiguration extracts telegram configuration from AlertManager config
+func (a *AlertCustomization) GetTelegramConfiguration(config string) map[string]interface{} {
+	var amConfig map[string]interface{}
+	if err := yaml.Unmarshal([]byte(config), &amConfig); err != nil {
+		return map[string]interface{}{"enabled": false}
+	}
+
+	receivers, ok := amConfig["receivers"].([]interface{})
+	if !ok {
+		return map[string]interface{}{"enabled": false}
+	}
+
+	var botTokens []string
+	var chatIDs []string
+
+	for _, r := range receivers {
```

Comment:
Would be good to avoid deep nesting for readability and maintainability.
```suggestion
	for _, r := range receivers {
	        receiver, ok := r.(map[string]interface{})
	        if !ok {
		        continue
	        }
        
	        telegramConfigs, exists := receiver["telegram_configs"]
	        if !exists {
		        continue
	        }
        
	        configList, ok := telegramConfigs.([]interface{})
	        if !ok {
		        continue
	        }
        
	        for _, config := range configList {
		        confMap, ok := config.(map[string]interface{})
		        if !ok {
			        continue
		        }
        
		        if token, ok := confMap["bot_token"]; ok {
			        botTokens = append(botTokens, fmt.Sprintf("%v", token))
		        }
        
		        if chat, ok := confMap["chat_id"]; ok {
			        chatIDs = append(chatIDs, fmt.Sprintf("%v", chat))
		        }
	        }
        }
```